geode-core/src/main/java/org/apache/geode/internal/cache/map/RegionMapPut.java

```diff
@@ -396,12 +410,35 @@ private boolean checkUninitializedRegionPreconditions() {
   private boolean checkCreatePreconditions() {
     if (isIfNew()) {
       if (!getRegionEntry().isDestroyedOrRemoved()) {
+        // retain the version stamp of the existing entry for use in processing failures
+        EntryEventImpl event = getEvent();
+        if (event.getRegion().getConcurrencyChecksEnabled() &&
+            event.getOperation() == Operation.PUT_IF_ABSENT &&
+            !event.hasValidVersionTag() &&
+            event.isPossibleDuplicate()) {
+          if (ValueComparisonHelper.checkEquals(getRegionEntry().getValue(),
+              getEvent().basicGetNewValue(),
+              isCompressedOffHeap(event), event.getRegion().getCache())) {
+            logger.info("retried putIfAbsent found same value already in cache "
+                + "- allowing the operation.  entry={}; event={}", getRegionEntry(), getEvent());
+            event.setPossibleDuplicate(true);
+            this.overwritePutIfAbsent = true;
+            return true;
+          }
+        }
         return false;
       }
     }
     return true;
   }
 
+
+  private boolean isCompressedOffHeap(EntryEventImpl event) {
+    return event.getContext() != null ? false
```

Comment:
I think this condition check can be removed as it is no need to distinguish a client cache or not.